### ISSUE TITLE
Hilt TestInstallIn and UninstallModules causing duplicate binding errors

### ISSUE THREAD

**Issue by alisonthemonster** [google/dagger#2726]
Hi there!

I'm using Hilt to inject some dependencies like so:
```
@Module
@InstallIn(SingletonComponent::class)
object SomethingModule {

    @Provides
    fun provideSomething(@ApplicationContext context: Context): Something {
        //return RealSomething
    }
}
```
I want to replace this binding in my tests and following this [guide][1] I added this fake module to the same folder as my instrumented test:
```
@Module
@TestInstallIn(
    components = [SingletonComponent::class],
    replaces = [SomethingModule::class]
)
object FakeSomethingModuleModule {

    @Provides
    fun provideSomethingModule(): Something {
        return FakeSomething()
    }
}
```

When I try to run my instrumented test the build fails with a duplicate binding error:
```
error: [Dagger/DuplicateBindings] com.example.Something is bound multiple times:
  public abstract static class SingletonC implements HiltWrapper_ActivityRetainedComponentManager_ActivityRetainedComponentBuilderEntryPoint,
```

Commenting out the Fake Module fixes the issue. Anyone see what I'm missing here?

I also tried the  `@UninstallModules` and ran into the exact same build error: `X is bound multiple times: public abstract static class SingletonC implements HiltWrapper_ActivityRetainedComponentManager_ActivityRetainedComponentBuilderEntryPoint`
```
@UninstallModules(SomethingModule::class)
@HiltAndroidTest
@RunWith(AndroidJUnit4::class)
class MyInstrumentedTest {

    @Module
    @InstallIn(SingletonComponent::class)
    object FakeSomethingModule {

        @Provides
        fun provideSomthing(): Something = FakeSomething()
    }
```

Do you see what I'm missing here? Will try and create a clean repro example in the mean time!

  [1]: https://developer.android.com/training/dependency-injection/hilt-testing#replace-binding

**Comment by alisonthemonster** [comment]
Wasn't able to create a clean repro example, but my project is open source so you can see exactly what is happening here:
https://github.com/alisonthemonster/Presently/pull/215

**Comment by alisonthemonster** [comment]
Going to close this as I figured it out! I believe there were a few issues in my code:

1. Make sure the fake implementations have @Inject constructor()

2. It wasn't apparent in this sample but in my actual code the module I was trying to test was included as a binding to another class using @Module(includes = SomethingBinding). I created a module just for the object I wanted to mock in my instrumented tests and replaced that with my test module and that did the trick!

Closing this out as my tests run now 🥳
